lms/lmsweb/views.py

```diff
@@ -130,6 +130,15 @@ def favicon():
     )
 
 
+@webapp.before_request
+def banned_page():
+    if (
+        request.endpoint not in ('login', 'logout')
```

Comment:
I think checking if `current_user.is_authenticated and current.user.role.is_banned` might be more accurate